### PR TITLE
feat(Code): Implementation Web & Native

### PR DESCRIPTION
**Pull request by saurabhdaware**
## Description

This PR adds Web and Native implementation for `Code` component.

**PR Stack**
- [x] API Decisions (#711)
- [x] Web and Native Implementation (this PR)
- [x] ~Storybook Docs Improvements (todo) (Basic storybook is added in this PR)~ (Added it in this PR itself)
- [x] Tests (#729)
- [ ] Add final export, changeset, and release component (todo)

### Android

image

### iOS
image

### Web

image

### CODE REVIEW

File: packages/blade/package.json [hunk 1]
```diff
@@ -55,6 +55,11 @@
     "react": "yarn run react:storybook",
     "react:storybook": "FRAMEWORK=REACT start-storybook -c ./.storybook/react -p 9009",
     "react:storybook:build": "FRAMEWORK=REACT build-storybook -c ./.storybook/react -o storybook-site",
+    "start:ios": "run-p react-native:storybook:start react-native:storybook:ios",
```
Comment (saurabhdaware) [inline packages/blade/package.json:58]: TODO: pull master to remove this

File: packages/blade/src/components/Typography/Code/Code.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+   *
+   * ```tsx
+   * // ❌ Bad
+   * <Code><Box>something</Box></Code>
+   * ```
+   *
+   *```tsx
+   * // ✅ Good
+   * <Code>something</Code>
+   * ```
```
Comment (anuraghazra) [inline packages/blade/src/components/Typography/Code/Code.tsx:19]: Don't think this is needed, since the TS error will be there anyway.

File: packages/blade/src/components/Typography/Code/Code.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  children: React.ReactNode;
+};
+
+const platformType = getPlatformType();
+const isPlatformWeb = platformType === 'browser' || platformType === 'node';
+
+const CodeContainer = styled(Box)<CodeContainerProps>((props) => {
+  const padding =
+    props.size === 'large'
+      ? `${props.theme.spacing[0]}px ${props.theme.spacing[2]}px`
```
Comment (anuraghazra) [inline packages/blade/src/components/Typography/Code/Code.tsx:47]: let's use the makeSpace utility

File: packages/blade/src/components/Typography/Code/Code.tsx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  return {
+    padding,
+    backgroundColor: props.theme.colors.brand.gray[300],
+    borderRadius: props.theme.border.radius.medium,
+    display: isPlatformWeb ? 'inline-block' : undefined,
+  };
+});
+
+/**
+ * Code component can be used for displaying token, variable names, or inlined code snippets.
```
Comment (anuraghazra) [inline packages/blade/src/components/Typography/Code/Code.tsx:58]: Nice.

File: packages/blade/src/components/Typography/Code/Code.tsx [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+ *   <Code>SENTRY_TOKEN</Code>
+ *   <Text> normal text</Text>
+ * </Box>
+ * ```
+ */
+const Code = ({ children, size = 'medium', type = 'subtle' }: CodeProps): JSX.Element => {
+  return (
+    <CodeContainer size={size} as={isPlatformWeb ? 'span' : undefined}>
+      <BaseText
+        color={`surface.text.${type ?? 'subtle'}.lowContrast`}
```
Comment (anuraghazra) [inline packages/blade/src/components/Typography/Code/Code.tsx:85]: Do we need to do `type ?? subtle`? 

We are already setting default value to type in the param.

File: packages/blade/src/components/Typography/Code/Code.tsx [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+   * Decides the visibility and color of Code
+   *
+   * @default subtle
+   */
+  type?: TextTypes;
+};
+
+type CodeContainerProps = {
+  size: CodeProps['size'];
+  children: React.ReactNode;
```
Comment (anuraghazra) [inline packages/blade/src/components/Typography/Code/Code.tsx:38]: 🤔 don't think you need to specify children for styled components.

File: packages/blade/src/components/Typography/Code/Code.tsx (hunk 5 above)
Comment (saurabhdaware) [reply packages/blade/src/components/Typography/Code/Code.tsx:85]: I did for TS. It was still thinking that value can be undefined 🤔

File: packages/blade/src/components/Typography/Code/Code.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+};
+
+const CodeTemplate: ComponentStory<typeof CodeComponent> = (args) => (
+  // For React Native, use flex to align items correctly
+  <Text>
+    Lorem ipsum normal text <CodeComponent {...args} /> component
+  </Text>
+);
+
+export default CodeStoryMeta;
```
Comment (anuraghazra) [inline packages/blade/src/components/Typography/Code/Code.stories.tsx:22]: Maybe we should add a kitchen sink example with all the variants.

File: packages/blade/src/components/Typography/Code/Code.tsx (hunk 6 above)
Comment (saurabhdaware) [reply packages/blade/src/components/Typography/Code/Code.tsx:38]: Oh right. Will remove it.

File: packages/blade/src/components/Typography/Code/Code.tsx [hunk 8]
```diff
@@ -0,0 +1,96 @@
+import styled from 'styled-components';
```
Comment (anuraghazra) [inline packages/blade/src/components/Typography/Code/Code.tsx:1]: Add index.ts and rexport the Code. 

And also reexport on Typography/index.ts

File: packages/blade/src/components/Typography/Code/Code.stories.tsx (hunk 7 above)
Comment (saurabhdaware) [reply packages/blade/src/components/Typography/Code/Code.stories.tsx:22]: Was thinking that but we don't really have kitchen sink for other typography components and also the variants can be changed from docs controller itself and they don't visibility look that different because it's just change in size.

File: packages/blade/src/components/Typography/Code/Code.tsx (hunk 8 above)
Comment (saurabhdaware) [reply packages/blade/src/components/Typography/Code/Code.tsx:1]: Was planning to do that at the end in followup PR. So that this PR can be merged after review. Added the todo for this in description

File: packages/blade/src/components/Typography/Code/Code.tsx (hunk 5 above)
Comment (anuraghazra) [reply packages/blade/src/components/Typography/Code/Code.tsx:85]: Don't think it's throwing error, maybe try restarting the ts server?

Comment (saurabhdaware) [reply packages/blade/src/components/Typography/Code/Code.tsx:85]: ohhh right my badd!! My earlier implementation was a bit different where there was in some nested component where it was throwing the error. You're right its not throwing error here. Will remove it.

File: packages/blade/src/components/Typography/Code/Code.stories.tsx (hunk 7 above)
Comment (anuraghazra) [reply packages/blade/src/components/Typography/Code/Code.stories.tsx:22]: Yes in fact we should add other kitchen sink too. 

My main reason of adding kitchen sink examples is to catch visual regressions with chromatic. 
If the storybook doesn't render the size and other variants chromatic won't screenshot them.

File: packages/blade/src/components/Typography/Code/Code.tsx (hunk 2 above)
Comment (saurabhdaware) [reply packages/blade/src/components/Typography/Code/Code.tsx]: done

File: packages/blade/src/components/Typography/Code/Code.tsx (hunk 3 above)
Comment (saurabhdaware) [reply packages/blade/src/components/Typography/Code/Code.tsx]: done

File: packages/blade/src/components/Typography/Code/Code.tsx (hunk 5 above)
Comment (saurabhdaware) [reply packages/blade/src/components/Typography/Code/Code.tsx]: done

File: packages/blade/src/components/Typography/Code/Code.tsx (hunk 6 above)
Comment (saurabhdaware) [reply packages/blade/src/components/Typography/Code/Code.tsx]: done

File: packages/blade/src/components/Typography/Code/Code.stories.tsx [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+  component: CodeComponent,
+  args: {
+    size: 'medium',
+    type: 'subtle',
+    children: 'SENTRY_AUTH_TOKEN',
+  },
+};
+
+const CodeTemplate: ComponentStory<typeof CodeComponent> = (args) => (
+  // For React Native, use flex to align items correctly
```
Comment (divyanshu013) [inline packages/blade/src/components/Typography/Code/Code.stories.tsx:16]: Didn't understand the context of this comment, seems unrelated 🤔

**Review by divyanshu013 — APPROVED** [review]
Looks good to me.

File: packages/blade/src/components/Typography/Code/Code.stories.tsx (hunk 9 above)
Comment (saurabhdaware) [reply packages/blade/src/components/Typography/Code/Code.stories.tsx:18]: As in that story will not look properly aligned on react native. For that we have to use flex and do it properly. I thought of conditionally adding a react native layout based on platform but might be overkill since storybook is mostly visited from web.

File: packages/blade/src/components/Typography/Code/Code.stories.tsx (hunk 7 above)
Comment (saurabhdaware) [reply packages/blade/src/components/Typography/Code/Code.stories.tsx:24]: Added the sink 👯

**Comment by divyanshu013** [comment]
You can also click on UI tests and review them. If you're happy with the results you may click approve on Chromatic.

image

**Review by chaitanyadeorukhkar — APPROVED** [review]

File: packages/blade/src/components/Typography/Code/Code.tsx [hunk 10]
```diff
@@ -66,13 +59,13 @@ const CodeContainer = styled(Box)<CodeContainerProps>((props) => {
...
-const Code = ({ children, size = 'medium', type = 'subtle' }: CodeProps): JSX.Element => {
+const Code = ({ children, size = 'small' }: CodeProps): JSX.Element => {
   return (
     <CodeContainer size={size} as={isPlatformWeb ? 'span' : undefined}>
       <BaseText
-        color={`surface.text.${type}.lowContrast`}
+        color="surface.text.subtle.lowContrast"
         fontFamily="code"
-        fontSize={size === 'large' ? 100 : 75}
+        fontSize={size === 'small' ? 75 : 100}
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Typography/Code/Code.tsx:68]: nit: create some function that takes in `size`, has a switch case or if..elseif and returns 75 or 100 depending on the size? This way if we add another size in the future, we will have a cleaner implementation for it instead of just checking for `small`?

**Review by chaitanyadeorukhkar — APPROVED** [review]
One feedback, otherwise lgtm

File: packages/blade/src/components/Typography/Code/Code.tsx (hunk 10 above)
Comment (saurabhdaware) [reply packages/blade/src/components/Typography/Code/Code.tsx]: done
